app/models/google_event.rb

```diff
@@ -0,0 +1,39 @@
+class GoogleEvent
+  class << self
+    #You can specify custom fields: https://developers.google.com/google-apps/calendar/v3/reference/events
+    FIELDS = "items(start, end, summary, recurrence, creator(displayName))".freeze
+
+    def list_events(credentials, starting, ending)
+      events = {}
+      rooms = ConferenceRoom.all
+      get_service(credentials).batch do |s|
+        rooms.each do |room|
+          s.list_events(room.email, fields: FIELDS, single_events: true, time_min: starting.rfc3339(9), time_max: ending.rfc3339(9), time_zone: 'Europe/Warsaw') do |n, _|
+            n&.items&.each do |e|
```

Comment:
Chaining `&.` is dangerous, I'd explicitly validate the service response and handle sad paths